### ISSUE TITLE
Error with checkpoint on evaluate.py and visualize.py

### ISSUE THREAD

**Issue by thepuremind** [sercant/mobile-segmentation#3]
Hey, @sercant!

In the case of running the file evaluate.py or visualize.py with your weights (shufflenetv2_dpc_cityscapes_71_3), the following error occurs:

For visualize.py:
NotFoundError (see above for traceback): Restoring from checkpoint failed. This is most likely due to a Variable name or other graph key that is missing from the checkpoint. Please ensure that you have not altered the graph expected based on the checkpoint. Original error:

Key aspp0/BatchNorm/beta not found in checkpoint
	 [[node save/RestoreV2 (defined at visualize.py:276) ]]
	 [[node save/RestoreV2 (defined at visualize.py:276) ]]

For evaluate.py:
NotFoundError (see above for traceback): Restoring from checkpoint failed. This is most likely due to a Variable name or other graph key that is missing from the checkpoint. Please ensure that you have not altered the graph expected based on the checkpoint. Original error:

Key aspp0/BatchNorm/beta not found in checkpoint
	 [[node save/RestoreV2 (defined at evaluate.py:162) ]]

**Comment by sercant** [comment]
Hello @thepuremind,

The model that you are trying is the **DPC** encoder head variant. I assume that this is due to a missing parameter that needs to be passed to the scripts. 

To fix this error pass this parameter to the script: 
```
--dense_prediction_cell_json=./core/dense_prediction_cell_branch5_top1_cityscapes.json
```

Let me know if it fixes the issue.  I am going to be updating the documentation in the following months. Until the documentation is updated I will keep the issue open.

**Comment by thepuremind** [comment]
@sercant , thanx!
It fixed this problem. Also I have a questions. How I can start to evaluate on test set of Cityscapes data? 
I think that need to add test set in tfrecord. How to do it?
What I need doing to run to training on other data?

**Comment by sercant** [comment]
@thepuremind, `test` set doesn't have the ground truth files provided. The evaluation should be done at [www.cityscapes-dataset.com](www.cityscapes-dataset.com). To send to the evaluation one can use the [visualize.py](visualize.py) to generate the output, zip and send to the server.

To generate the `test` tfrecord files:
1. Modify the line https://github.com/sercant/mobile-segmentation/blob/1ca269eba08a6343aa7e9a9f61488143b92f9316/dataset/build_cityscapes_data.py#L169 and add `'test'` to the list.
2. Run the build script again.
3. To visualize the `test` set, pass parameter `--vis_split=test` to the [visualize.py](visualize.py).

To run on your own dataset, please examine the scripts under the [dataset](dataset) folder and generate your own tfrecord files. Then add the information of the dataset to [segmentation_dataset.py](dataset/segmentation_dataset.py) file. As there are already several examples within the file, it should be straight forward.
